includes/data.php

```diff
@@ -778,6 +778,45 @@ function pods_v( $var = null, $type = 'get', $default = null, $strict = false, $
 					}
 				}
 				break;
+			case 'post_id':
+				if ( is_array( $var ) ) {
+					if ( isset( $var['post_id'] ) ) {
+						$post_id = $var['post_id'];
+					}
+					if ( isset( $var['post_type'] ) ) {
+						$post_type = $var['post_type'];
+					} else {
+						$post_type = 'post';
+					}
+				} elseif ( null === $var ) {
+					$post_id = get_the_ID();
+					$post_type = get_post_type( $post_id );
+				} else {
+					$post_id   = $var;
+					$post_type = 'post';
+				}
+				if ( defined( 'ICL_SITEPRESS_VERSION' ) ) {
+					/* Only call filter if WPML is installed */
+					$post_id = apply_filters( 'wpml_object_id', $post_id, $post_type, true );
+				}
+				// Add other translation plugin specific code here
+
+				/**
+				 * Filter to override post_id
+				 *
+				 * Generally used with language translation plugins in order to return the post id of a
+				 * translated post
+				 *
+				 * @param  int $post_id The post ID of current post
+				 * @param  string $post_type Post type of current post
+				 * @param  mixed $default The default value to set if variable doesn't exist
+				 * @param  bool $strict Only allow values (must not be empty)
+				 * @param  array $params Set 'casting'=>true to cast value from $default, 'allowed'=>$allowed to restrict a value to what's allowed
+				 *
+				 * @since 2.6.6
+				 */
+				$output = apply_filters( 'pods_var_post_id', $post_id, $post_type, $default, $var, $strict, $params );
```

Comment:
Thinking about it more, do we really even need to pass the $post_type in through the array?  We can just always call get_post_type() on the id and it should be returned.